h5netcdf/core.py

```diff
@@ -724,6 +748,21 @@ def _create_child_variable(
         else:
             self._root._check_valid_netcdf_dtype(dtype)
 
+        # compound type handling
+        if np.dtype(dtype).kind == "c":
+            if not isinstance(dtype, CompoundType) and self._root._auto_complex:
+                # todo check compound types for
+                #  existing complex types which may be used
+                itemsize = np.dtype(dtype).itemsize
+                if itemsize == 8:
+                    width = "FLOAT"
+                elif itemsize == 16:
+                    width = "DOUBLE"
```

Comment:
I don't know how common other type sizes are, but it's probably wise to explicitly error here

Reply:
`width = {8: "FLOAT", 16: "DOUBLE"}[itemsize]` still feels like a safer way to write this tp me :)